docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,515 @@
+
+  if (username == null || password == null) {
+    return Response(statusCode: HttpStatus.badRequest);
+  }
+
+  final authenticator = context.read<Authenticator>();
+
+  final user = authenticator.findByUsernameAndPassword(
+    username: username,
+    password: password,
+  );
+
+  if (user == null) {
+    return Response(statusCode: HttpStatus.unauthorized);
+  } else {
+    return Response.json(
+      body: { 'token': username },
+    );
+  }
+}
+```
+
+To people familiar with Dart Frog, the code above should be no real challenge, we are simply handling the
+request in the following steps:
+
+- Check if we have all the info needed, returning `badRequest` otherwise.
+- Get our `Authenticator` dependency from our dependency injection.
+- User the authenticator to get a user that match the request's credential.
+- Returns `unauthorized` (401) if there is no user, or returns the user username as the authentication token otherwise.
+
+But wait, you could be thinking that using the user username as an authentication token is quite unsafe.
+And for sure it is, but for now, let's just for the sake of simplicity, for now lets go with that
+in order to finish our authentication setup before introducing more complex security methods.
+
+Try now running a `curl` in the terminal to get a token:
+
+```bash
+curl -X POST \
+     -H "Content-Type: application/json" \
+     -d '{"username": "john", "password": "123"}'  \
+     http://localhost:8080/sign_in
+
+# {"token":"john"}
+```
+
+## Requiring authentication to access routes
+
+Now that we have the means to get an authentication token, we can now protect routes to require a token
+to be provided in
+order to access them.
+
+To start lets create the following route:
+
+```dart
+// routes/tasks/index.dart
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+
+Future<Response> onRequest(RequestContext context) {
+  return switch (context.request.method) {
+    HttpMethod.post => _onPost(context),
+    _ => Future.value(Response(statusCode: HttpStatus.methodNotAllowed)),
+  };
+}
+
+Future<Response> _onPost(RequestContext context) async {
+  final task = await context.request.body();
+  return Response.json(
+    body: {
+      'recorded_task': task,
+    },
+  );
+}
+```
+
+This route doesn't do much for now, it just reads the request and answers with a body with the
+received task name. Also it isn't yet protected.
+
+To protect our route, we will use [`dart_frog_auth`](https://pub.dev/packages/dart_frog_auth), a
+package provided by Dart Frog that makes it easier to implement token based authentications.
+
+So lets start by adding it to the project:
+
+```bash
+dart pub add dart_frog_auth
+```
+
+First, we need to create a method in our authenticator class which will be responsible to validate
+an authentication token, since right now, our authentication token is just the user's username,
+we can add the following snippet to the `Authenticator` class:
```

Comment:
```suggestion
First, we need to create a method in our `Authenticator` class which will be responsible of validating
an authentication token. Since for now, our authentication token is just the user's username,
we can add the following snippet to the `Authenticator` class:
```